do/csi/driver/controller.go

```diff
@@ -3,48 +3,347 @@ package driver
 import (
 	"context"
 	"errors"
+	"fmt"
+	"net/http"
+	"strconv"
+	"strings"
 
 	csi "github.com/container-storage-interface/spec/lib/go/csi/v0"
+	"github.com/digitalocean/godo"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
 )
 
-// CreateVolume creates a new volume
-func (d *Driver) CreateVolume(context.Context, *csi.CreateVolumeRequest) (*csi.CreateVolumeResponse, error) {
-	return nil, errors.New("not implemented")
+const (
+	_  = iota
+	KB = 1 << (10 * iota)
+	MB
+	GB
+	TB
+)
+
+const (
+	defaultVolumeSizeInGB = 16 * GB
+
+	createdByDO = "Created by DigitalOcean CSI driver"
```

Comment:
Do you know if there's any other metadata we can add here that would be passed into the CSI driver (like cluster name, etc)?

Reply:
Asking cause checking this description doesn't work if there are multiple clusters using this CSI driver.

Reply:
The description is only checked if the driver is created by the CSI driver or not. We don't use it for anything else. For multiple K8S clusters, you're right. There is a way to pass down metadata inside the request:

```
	// Plugin specific parameters passed in as opaque key-value pairs.
	// This field is OPTIONAL. The Plugin is responsible for parsing and
	// validating these parameters. COs will treat these as opaque.
	Parameters map[string]string `protobuf:"bytes,4,rep,name=parameters" json:"parameters,omitempty" protobuf_key:"bytes,1,opt,name=key" protobuf_val:"bytes,2,opt,name=value"`
```

However it's not clear to me what the CO (in our case Kubernetes) passes down. I'll be make sure to check it and incorporate any information into the description (such as including the cluster name, if available)